### PR TITLE
Add __contains__ to TextChannel, DMChannel, Guild

### PR DESCRIPTION
**Pull request by sevenc-nanashi**
## Summary

I added `__contains__` to TextChannel, DMChannel, Guild.
### TextChannel, DMChannel
checks message is in the channel.
image
image
Channels are like this:
image

### Guild
checks channel is in the guild.
image
image

(Sorry for light theme and japanese)
## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: discord/channel.py [hunk 1]
```diff
@@ -1231,6 +1246,13 @@ def __init__(self, *, me, state, data):
         self.me = me
         self.id = int(data['id'])
 
+    async def __contains__(self, item):
```
Comment (AbstractUmbra) [inline discord/channel.py:1249]:
```suggestion
    def __contains__(self, item):
```
Not sure why this is coroutine?

Comment (sevenc-nanashi) [reply discord/channel.py]: Just my miss, fixed

File: discord/channel.py [hunk 2]
```diff
@@ -69,6 +69,10 @@ class TextChannel(discord.abc.Messageable, discord.abc.GuildChannel, Hashable):
 
             Returns the channel's name.
 
+        .. describe:: y in x
+
+            Checks if message is in the channel.
+
```
Comment (TheOneMusic) [inline discord/channel.py:75]:
```suggestion
.. versionadded:: 2.0
```

File: discord/channel.py [hunk 3]
```diff
@@ -1211,6 +1222,10 @@ class DMChannel(discord.abc.Messageable, Hashable):
 
             Returns a string representation of the channel
 
+        .. describe:: y in x
+
+            Checks if message is in the channel.
+
```
Comment (TheOneMusic) [inline discord/channel.py:1228]:
```suggestion
            .. versionadded:: 2.0
```

File: discord/guild.py [hunk 4]
```diff
@@ -73,6 +73,10 @@ class Guild(Hashable):
 
             Checks if two guilds are not equal.
 
+        .. describe:: y in x
+
+            Checks if channel is in the guild.
+
```
Comment (TheOneMusic) [inline discord/guild.py:79]:
```suggestion
            .. versionadded:: 2.0
```

**Comment by Rapptz** [comment]
Sorry, I don't think abusing `__contains__` like this makes a lot of sense to me.

**Pull request closed without merging**